plugin/PDFView.qml

```diff
@@ -120,9 +121,61 @@ SilicaFlickable {
             NumberAnimation { id: scaleOut; property: "scale"; duration: 200; to: 1.; easing.type: Easing.InOutCubic }
         }
     }
+    NumberAnimation {
+        id: selectionOffset
+        property real start
+        duration: 200
+        easing.type: Easing.InOutCubic
+        target: base
+        property: "contentY"
+    }
 
     // Ensure proper zooming level when device is rotated.
     onWidthChanged: adjust()
+    Component.onCompleted: {
+        // Avoid hard dependency to feedback
+        _feedbackEffect = Qt.createQmlObject("import QtQuick 2.0; import QtFeedback 5.0; ThemeEffect { effect: ThemeEffect.PressWeak }",
+                                             base, 'ThemeEffect')
+        if (_feedbackEffect && !_feedbackEffect.supported) {
+            _feedbackEffect = null
+        }
+    }
+
+    PDF.Selection {
+        id: selection
+        
+        property bool dragging: drag1.press || drag2.press
+        property bool selected: count > 0
+
+        canvas: pdfCanvas
+        wiggle: Theme.itemSizeSmall / 2
+
+        onDraggingChanged: {
+            if (dragging) {
+                if (!selectionOffset.running)
+                    selectionOffset.start = base.contentY
+
+                // Limit offset when being at the bottom of the view.
+                selectionOffset.to = selectionOffset.start +
+                    Math.min(Theme.itemSizeSmall
+                             , Math.max(0, base.itemHeight - base.height - base.contentY))
+                // Limit offset when being at the top of screen
+                selectionOffset.to = Math.max(
+                    base.contentY
+                    , Math.min(selectionOffset.to
```

Comment:
Ok, got confused with the style for logical operators or inline test that should be written:
```
 cond1
 || cond2
```
and
```
test
? if-action
: else-action
```
when lines are too longs, if I'm not mistaken.

Anyway, I've corrected this comma style issue in PDFSelectionDrag.qml and PDFSelectionHandle.qml also.